server-state-machine.go

```diff
@@ -591,6 +621,27 @@ func (state serverStateNegotiated) Next(_ handshakeMessageReader) (HandshakeStat
 			return nil, nil, AlertInternalError
 		}
 	}
+	if state.Params.UsingRawPublicKeys {
+		logf(logTypeHandshake, "[server] sending ClientCertType extension")
```

Comment:
BUG: You are violating a spec requirement here:
```   If the server does not send a certificate_request payload (for
   example, because client authentication happens at the application
   layer or no client authentication is required) or none of the
   certificates supported by the client (as indicated in the
   client_certificate_type extension in the client hello) match the
   server-supported certificate types, then the client_certificate_type
   payload in the server hello MUST be omitted.
```